### PR TITLE
Asset/AssetList improve helper methods

### PR DESCRIPTION
**Pull request by haakonvt**
### Changed
- Helper methods to get related resources on `Asset` class now accepts `asset_ids` as part of keyword arguments.
### Added
- Helper methods to get related resources on `AssetList` class now accept keyword arguments that are passed on to
  the list endpoint (for server-side filtering).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,13 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [7.14.0] - 2024-01-22
+### Changed
+- Helper methods to get related resources on `Asset` class now accepts `asset_ids` as part of keyword arguments.
+### Added
+- Helper methods to get related resources on `AssetList` class now accept keyword arguments that are passed on to
+  the list endpoint (for server-side filtering).
```
Comment (haakonvt) [inline CHANGELOG.md:25]:
```suggestion
- Helper methods to get related resources on `Asset` class now accept `asset_ids` as part of keyword arguments.
### Added
- Helper methods to get related resources on `AssetList` class now accept keyword arguments that are passed on to
  the list endpoint (for server-side filtering).
```

File: cognite/client/utils/_auxiliary.py [hunk 2]
```diff
@@ -177,6 +177,11 @@ def split_into_n_parts(seq: Sequence[T], *, n: int) -> Iterator[Sequence[T]]:
     yield from (seq[i::n] for i in range(n))
 
 
+@overload
+def split_into_chunks(collection: set, chunk_size: int) -> list[list]:
+    ...
+
+
 @overload
 def split_into_chunks(collection: list, chunk_size: int) -> list[list]:
```
Comment (doctrino) [inline cognite/client/utils/_auxiliary.py:186]: Can't this be added to the overload below? 

```python
def split_into_chunks(collection: list | set, chunk_size: int) -> list[list]:
```

File: cognite/client/data_classes/assets.py [hunk 3]
```diff
@@ -265,52 +265,58 @@ def subtree(self, depth: int | None = None) -> AssetList:
...
         Args:
-            **kwargs (Any): All extra keyword arguments are passed to time_series/list. NB: 'asset_ids' can't be used.
+            **kwargs (Any): All extra keyword arguments are passed to time_series/list.
         Returns:
             TimeSeriesList: All time series related to this asset.
         """
-        assert self.id is not None
-        return self._cognite_client.time_series.list(asset_ids=[self.id], **kwargs)
+        asset_ids = self._prepare_asset_ids("time series", kwargs)
+        return self._cognite_client.time_series.list(asset_ids=asset_ids, **kwargs)
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/data_classes/assets.py:281]: It's just merged in. That reminds me, I should do a de-duplication just to be sure

File: cognite/client/utils/_auxiliary.py (hunk 2 above)
Comment (haakonvt) [reply cognite/client/utils/_auxiliary.py:186]: Aye, yeah, will change

File: cognite/client/data_classes/assets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/data_classes/assets.py:281]: `remove_duplicates_keep_order([self.id, *user_kwargs.pop("asset_ids", [])])`

**Pull request merged**